PageObjects/CartPage.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.Playwright;
+using NUnit.Framework;
+using TechTalk.SpecFlow;
+
+namespace HW18_SpecFlow.PageObjects
+{
+    [Binding]
+    internal class CartPage
+    {
+        private readonly IPage page;
+
+        public CartPage(IPage page)
+        {
+            this.page = page;
+        }
+
+        public async Task VerifyPageUrl(string testPageUrl)
+        {
+            try
+            {
+                await page.WaitForURLAsync(testPageUrl);
+            }
+            catch (PlaywrightException e)
+            {
+                if (e.Message.Contains("crash"))
+                {
+                    Console.WriteLine("Page crashed: " + e.Message);
+                }
+            }
+        }
+
+        public async Task VerifyHeadingVisible(string heading)
+        {
+            var headingLocator = page.GetByRole(AriaRole.Heading, new() { Name = heading });
+            await Assertions.Expect(headingLocator).ToBeVisibleAsync();
+        }
+
+        public async Task VerifyProductAddedToCart(string addProductName)
+        {
+            var productTitleLocator = "//*[contains(@class, 'prod-title')]";
+            var allProducts = await page.Locator(productTitleLocator).AllInnerTextsAsync();
+            var productsList = allProducts.ToList();
+            Assert.That(productsList.Count, Is.GreaterThan(0), $"Products by locator {productTitleLocator} not found in the cart");
+
+            bool isAnyContainProductValue = productsList.Any(product => product.ToLower().Contains(addProductName.ToLower()));
+            Assert.That(isAnyContainProductValue, Is.True, $" No product '{addProductName}' in the cart");
+        }
+
+        public async Task RemoveProductFromCart(string removeProductName)
+        {
+            var productRowLocator = "//*[contains(@class, 'cart-product row')]";
+            var removeBtnLocator = "//*[starts-with(@class, 'remove-from-cart')]";
+            var products = await page.QuerySelectorAllAsync(productRowLocator);
+            Assert.That(products, Is.Not.Empty, "No products in the cart");
+
+            foreach (var product in products)
```

Comment:
this should be much simpler. Better to use approach which I shown on "25. Static class. CSS/xPath selectors. Playwright Trace viewer" -  how to work with tables or "Filtering Locators" approach from https://playwright.dev/dotnet/docs/locators